sigauth/signature.go

```diff
@@ -38,17 +40,25 @@ import (
 type SignatureAuth struct {
 	serviceRegManager *authservice.ServiceRegManager
 
-	serviceKey *rsa.PrivateKey
+	serviceKey authutils.PrivateKey
 }
 
 // Sign generates and returns a signature for the provided message
 func (s *SignatureAuth) Sign(message []byte) (string, error) {
-	hash, err := authutils.HashSha256(message)
-	if err != nil {
-		return "", fmt.Errorf("error hashing message: %v", err)
-	}
+	var opts crypto.SignerOpts
+	var err error
 
-	signature, err := rsa.SignPSS(rand.Reader, s.serviceKey, crypto.SHA256, hash, nil)
+	switch s.serviceKey.(type) {
+	case *rsa.PrivateKey, *ecdsa.PrivateKey:
+		message, err = authutils.HashSha256(message)
+		if err != nil {
+			return "", fmt.Errorf("error hashing message: %v", err)
+		}
+		opts = &rsa.PSSOptions{Hash: crypto.SHA256}
+	case ed25519.PrivateKey:
+		opts = &rsa.PSSOptions{Hash: 0}
```

Comment:
Why are we using `rsa` options for all of these key types?

Reply:
I could not find options for the other key types. In this case, it does not really matter, since the options struct is only used to specify the digest hash function.

Reply:
We are now using the `jwt.SigningMethod.Sign` function here instead of signing explicitly, so this should no longer be an issue.